Sources/DeepLearning/Operators/Basic.swift

```diff
@@ -0,0 +1,710 @@
+// Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import TensorFlowCore
+
+//===------------------------------------------------------------------------------------------===//
+// Shape Transformations
+//===------------------------------------------------------------------------------------------===//
+
+public extension TensorFlowScalar {
+    /// Convert to a tensor with the specified rank, with all dimensions equal to 1.
+    @inlinable
+    func makeTensor(rank: Int) -> Tensor<Self> {
+        return Tensor(repeating: self, shape: TensorShape(rank))
+    }
+}
+
+public extension Tensor {
+    /// Unpacks the given dimension of a rank-`R` tensor into multiple rank-`(R-1)` tensors. Unpacks 
+    /// `N` tensors from this tensor by chipping it along the `axis` dimension, where `N` is
+    /// inferred from this tensor's shape. For example, given a tensor with shape `[A, B, C, D]`:
+    /// 
+    ///   - If `axis == 0` then the `i`-th tensor in the returned array is the slice 
+    ///     `self[i, :, :, :]` and each tensor in that array will have shape `[B, C, D]`. 
+    ///     (Note that the dimension unpacked along is gone, unlike
+    ///     `Tensor.split(numSplits:alongAxis)`, or `Tensor.split(sizes:alongAxis)`).
+    ///   - If `axis == 1` then the `i`-th tensor in the returned array is the slice 
+    ///     `value[:, i, :, :]` and each tensor in that array will have shape `[A, C, D]`.
+    ///   - Etc.
+    ///
+    /// This is the opposite of `Tensor.init(stacking:alongAxis:)`.
+    ///
+    /// - Parameters:
+    ///   - axis: Dimension along which to unstack. Negative values wrap around.
+    /// 
+    /// - Precondition: `axis` must be in the range `[-rank, rank)`, where `rank` is the rank of the
+    ///   provided tensors.
+    /// 
+    /// - Returns: Array containing the unstacked tensors.
+    @inlinable
+    @differentiable(vjp: _vjpUnstack(alongAxis:) where Scalar: TensorFlowFloatingPoint)
+    func unstack(alongAxis axis: Int = 0) -> [Tensor] {
```

Comment:
Non-mutating methods should read like a noun instead of an imperative verb phrase.
```suggestion
    func unstacked(alongAxis axis: Int = 0) -> [Tensor] {
```